### PR TITLE
add noise detector and optimized canny edge detection

### PR DESCRIPTION
**Pull request by jmoxrox**

### CODE REVIEW

File: UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import numpy
 
 CIRCLE_SCORE_THRESHOLD = 0.45 #needs to be optomized
+NOISE_SCORE_THRESHOLD = 0.6
 SQUARE_RECTANGLE_EIGEN_RATIO_THRESHOLD = 1.5 #not tested
 PENTAGON_STAR_CLUSTER_THRESHOLD = 8 #not tested
 OCTAGON_CROSS_CLUSTER_THRESHOLD = 10 #not tested
```
Comment (jvillemare) [inline UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py]: [There should be spaces in between the `#` and the text of the inline comment, per PEP8 standards.](https://www.python.org/dev/peps/pep-0008/#inline-comments)

Comment (jvillemare) [reply UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py]: Additionally, should these variables be moved to a `settings.py` file, @DarkmatterVale ?

File: UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py [hunk 2]
```diff
@@ -34,14 +35,14 @@ def load_polar_side_counter(self):
         self.polar_side_minimums = self.polar_side_counter.get_polar_side_minimums()
         self.num_polar_side_minimums = len(self.polar_side_counter.get_polar_side_minimums())
         self.circle_score = self.polar_side_counter.get_circle_score()
+        self.noise_score = self.polar_side_counter.get_noise_score()
         self.origin = self.polar_side_counter.get_origin()
 
     def determine_shape(self):
```
Comment (jvillemare) [inline UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py:31]: This function and all other functions should have documentation in the form of:

```
def determine_shape(self):
    """
    Function description.

    :param x:      Description of variable,
    :type x:       Type of variable,

    :return:       What the function returns.
    """
```

File: UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/tester.py [hunk 3]
```diff
@@ -4,7 +4,8 @@
 from PIL import Image
```
Comment (jvillemare) [inline UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/tester.py:4]: What I heard from John, @DarkmatterVale , is that this PR is to get code necessary for @Inffzy . I believe he will remove this `tester.py` file at a later time.

**Review by jvillemare — CHANGES_REQUESTED** [review]

File: UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py (hunk 1 above)
Comment (DarkmatterVale) [reply UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py]: Yea I agree.

**Review by DarkmatterVale — CHANGES_REQUESTED** [review]

**Review by DarkmatterVale — APPROVED** [review]
LGTM

File: UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py (hunk 2 above)
Comment (jvillemare) [reply UpdatedImageProcessing/UpdatedImageProcessing/ShapeDetection/shape_classification.py:31]: For the future, if you have functions like `determine_shape()` that are void (don't return anything) and don't take any inputs, you can just describe them as:

```
def determine_shape(self);
    """
    Function description.

    :return:       nothing.
    """
```

It's just important to describe what the function does for posterity.

**Pull request merged**